lib/webdriver-plus.ts

```diff
@@ -113,26 +121,48 @@ class WebElementRect implements ClientRect {
   get right(): number { return this.rect.x + this.rect.width; }
 }
 
-async function findContentHelper(driver: WebDriver, finder: WebElement|null,
-                                 selector: string, contentRE: RegExp): Promise<WebElement> {
+async function findContentHelper(driver: WebDriver, finder: WebElement|null, selector: string,
+                                 contentRE: RegExp, throwErrOnMiss: boolean = true): Promise<WebElement> {
   // tslint:disable:no-shadowed-variable
   try {
-    return await driver.executeScript<WebElement>( () => {
+    return await driver.executeScript<WebElement>(() => {
       const finder = (arguments[0] || window.document);
       const elements = [...finder.querySelectorAll(arguments[1])];
       const contentRE = new RegExp(arguments[2]);
+      const throwErrOnMiss = arguments[3];
       const found = elements.find((el) => contentRE.test(el.innerText));
-      if (!found) { throw new Error(`None of ${elements.length} elements match ${contentRE}`); }
+      if (!found && throwErrOnMiss) { throw new Error(`None of ${elements.length} elements match ${contentRE}`); }
       return found;
-    }, finder, selector, contentRE.source);
+    }, finder, selector, contentRE.source, throwErrOnMiss);
   } catch (err) {
-    if (/None of .* elements match/.test(err)) {
-      throw new error.NoSuchElementError(err.message);
-    }
-    throw err;
+    return rethrowErrorAsSpecial(err);
   }
 }
 
+async function findClosestHelper(driver: WebDriver, finder: WebElement, selector: string): Promise<WebElement> {
+  // tslint:disable:no-shadowed-variable
+  try {
+    return await driver.executeScript<WebElement>(() => {
+      const finder = arguments[0];
+      const selector = arguments[1];
+      const elem = finder.closest(selector);
+      if (!elem) { throw new Error(`None of the ancestor elements match ${selector}`); }
```

Comment:
Same here: seems better to just return `elem` even if its falsy, and check the return value of `executeScript` to decide whether to throw `NoSuch...`. I imagine you'd need to change type param to `executeScript<WebElement|null>`